### PR TITLE
Add tolist to Paddle frontend

### LINKED ISSUE #17948: tolist

### PR DESCRIPTION
**Pull request by fadhlanzaky**
Close #17948

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -328,3 +328,10 @@ def fmin(self, y, name=None):
     def minimum(self, y, name=None):
         y_ivy = _to_ivy_array(y)
         return ivy.minimum(self._ivy_array, y_ivy)
+
+    @with_supported_dtypes(
+        {"2.5.0 and below": ("float32", "float64", "bool", "int8", "int32", "int64")},
+        "paddle",
+    )
```
Comment (MahmoudAshraf97) [inline ivy/functional/frontends/paddle/tensor/tensor.py:335]: this is not necessary since this function supports all dtypes

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 1 above)
Comment (fadhlanzaky) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: the decorator has been removed, also fixed some conflicts. please re-review the changes

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 2]
```diff
@@ -2586,3 +2586,34 @@ def test_paddle_instance_sgn(
...
+
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="tolist",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+    ),
+)
+def test_paddle_tolist(
```
Comment (MahmoudAshraf97) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:2598]:
```suggestion
def test_paddle_instance_tolist(
```

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Comment by MahmoudAshraf97** [comment]
for some reason the tests are failing for all backends so I'd request @AnnaTz review or please assign someone to investigate, Thanks Anna.
```python
E       AssertionError: The length of results from backend paddle and ground truthframework paddle does not match
E       
E       len(ret_np_flat) != len(ret_np_from_gt_flat):
E       
E       ret_np_flat:
E       
E       [array(0)]
E       
E       ret_np_from_gt_flat:
E       
E       []
E       Falsifying example: test_paddle_instance_tolist(
E           frontend='paddle',
E           on_device='cpu',
E           dtype_and_x=(['int16'], [array([0], dtype=int16)]),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E           ),
E           init_flags=FrontendMethodTestFlags(
E               num_positional_args=1,
E               as_variable=[False],
E               native_arrays=[False],
E           ),
E           frontend_method_data=FrontendMethodData(ivy_init_module=<module 'ivy.functional.frontends.paddle' from '/workspaces/ivy/ivy/functional/frontends/paddle/__init__.py'>, framework_init_module=<module 'paddle' from '/opt/fw/paddle/paddle/__init__.py'>, init_name='to_tensor', method_name='tolist'),
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.80.0', b'AXicY2BkQAUAACEAAg==') as a decorator on your test case

ivy_tests/test_ivy/helpers/assertions.py:117: AssertionError
```

**Comment by AnnaTz** [comment]
> for some reason the tests are failing for all backends so I'd request @AnnaTz review or please assign someone to investigate, Thanks Anna.
> 
> ```python
> E       AssertionError: The length of results from backend paddle and ground truthframework paddle does not match
> E       
> E       len(ret_np_flat) != len(ret_np_from_gt_flat):
> E       
> E       ret_np_flat:
> E       
> E       [array(0)]
> E       
> E       ret_np_from_gt_flat:
> E       
> E       []
> E       Falsifying example: test_paddle_instance_tolist(
> E           frontend='paddle',
> E           on_device='cpu',
> E           dtype_and_x=(['int16'], [array([0], dtype=int16)]),
> E           method_flags=FrontendMethodTestFlags(
> E               num_positional_args=0,
> E               as_variable=[False],
> E               native_arrays=[False],
> E           ),
> E           init_flags=FrontendMethodTestFlags(
> E               num_positional_args=1,
> E               as_variable=[False],
> E               native_arrays=[False],
> E           ),
> E           frontend_method_data=FrontendMethodData(ivy_init_module=<module 'ivy.functional.frontends.paddle' from '/workspaces/ivy/ivy/functional/frontends/paddle/__init__.py'>, framework_init_module=<module 'paddle' from '/opt/fw/paddle/paddle/__init__.py'>, init_name='to_tensor', method_name='tolist'),
> E       )
> E       
> E       You can reproduce this example by temporarily adding @reproduce_failure('6.80.0', b'AXicY2BkQAUAACEAAg==') as a decorator on your test case
> 
> ivy_tests/test_ivy/helpers/assertions.py:117: AssertionError
> ```

Hi both, there is an issue in our frontend method testing pipeline. I will open a PR to fix this shortly. Once it's been approved by @CatB1t and merged we can merge this PR as well.

**Review by AnnaTz — APPROVED** [review]
Actually, I ran this function's test locally after fixing the testing pipeline bug and it passes, so there is no reason to keep the PR open!

**Pull request merged**